hazelcast/src/main/java/com/hazelcast/nio/tcp/AbstractIOSelector.java

```diff
@@ -102,7 +102,16 @@ private void processSelectionQueue() {
             if (runnable == null) {
                 return;
             }
-            runnable.run();
+            if (runnable instanceof WriteHandler) {
```

Comment:
Can you add some documentation the the AbstractIOSelector class explaining the mechanism of migrating a WriteHandler? Currently I see loose parts, but it is difficult to understand the whole mechanism. And this is d*mn tricky stuff :)